src/processing.py

```diff
@@ -1,12 +1,85 @@
 from pathlib import Path
 import time
+import paramiko
+import os
+import base64
+import tempfile
+import shutil
 
-from .supabase import use_client, login
+from fastapi import HTTPException
+
+from .supabase import use_client, login, verify_token
 from .settings import settings
-from .models import StatusEnum, Dataset, QueueTask, Cog
+from .models import StatusEnum, Dataset, QueueTask, Cog, Thumbnail
 from .logger import logger
 from . import monitoring
 from .deadwood.cog import calculate_cog
+from .deadwood.thumbnail import calculate_thumbnail
+
+
+def pull_file_from_storage_server(remote_file_path: str, local_file_path: str):
+    # Check if the file already exists locally
+    if os.path.exists(local_file_path):
+        print(f"File already exists locally at: {local_file_path}")
+        return
+
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    print(
+        "connecting to:",
+        settings.storage_server_ip,
+        settings.storage_server_username,
+        settings.storage_server_password,
```

Comment:
do not print out passwords. 

This ends up in ipython logs, which are plain text and in case of docker can easily be read from the container history.